assets/js/src/url.js

```diff
@@ -0,0 +1,40 @@
+/* globals S */
+
+import h from './helpers';
+
+export default (function () {
+  'use strict';
+
+  // DOM queries
+  var headings = $('.chapter:not(.toc) > h1[id]');
+  // Internal variables
+  var offset = 50;
+  var headingsOffset = headings.map(function(el) {
+    return [h.getOffset(el), el.id];
+  });
+
+  var updateURL = function () {
```

Comment:
I wonder whether we should make this part of sidebar.js since they are highly similar?